src/main/entities/StudyPool.java

```diff
@@ -0,0 +1,120 @@
+package entities;
+
+
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Map;
+
+public class StudyPool {
+    /**
+     * The collection of all studies in the system.
+     * This is outside the Study class because it is a static collection of all studies.
+     * The StudyPool composes the Study class.
+     * The StudyPool class is a singleton class. This means that there is only one instance of the class.
+     */
+
+
+    /*
+    The collection of all studies in the system.
+    The key is the study's ID.
+    The value is the study.
+     */
+    private Map<Integer, Study> studies;
+
+
+    /**
+     * The constructor for the StudyPool class.
+     *
+     * @param studies a map of all studies in the system. The key is the study's ID. The value is the study.
+     */
+    public StudyPool(Map<Integer, Study> studies) {
+        this.studies = studies;
+    }
+
+
+    /**
+     * Returns the study with the given ID.
+     *
+     * @param studyID the ID of the study to be returned.
+     * @return the study with the given ID.
+     */
+    public Study getStudy(int studyID) {
+        return studies.get(studyID);
+    }
+
+
+    /**
+     * Check if the study with the given ID exists.
+     *
+     * @param studyID the ID of the study to be checked.
+     * @return true if the study exists, false otherwise.
+     */
+    public boolean studyExists(int studyID) {
+        return studies.containsKey(studyID);
+    }
+
+
+    /**
+     * Checks the type of the study with the given study ID.
+     *
+     * @param studyID the ID of the study to be checked.
+     * @return the type of the study if the given study ID exists, null otherwise.
+     */
+    public String checkStudyType(int studyID) {
+        if (this.studies.containsKey(studyID)) {
+            Study study = this.studies.get(studyID);
+            return study.getStudyType();
+        }
+        return null;
+    }
+
+
+    /**
+     * Adds a study to the study pool.
+     *
+     * @param study the study to be added.
+     */
+    public void addStudy(Study study) {
+        this.studies.put(study.getId(), study);
```

Comment:
What if it's duplicated?

Reply:
This is unlikely because a study has a unique id. If the added study has the same ID it is still the original study in the pool. Otherwise, it will be a new study added to the pool.